internal/venice-client-common/src/main/java/com/linkedin/venice/schema/GeneratedSchemaID.java

```diff
@@ -0,0 +1,53 @@
+package com.linkedin.venice.schema;
+
+import static com.linkedin.venice.schema.SchemaData.INVALID_VALUE_SCHEMA_ID;
+
+
+public class GeneratedSchemaID {
```

Comment:
At first look, I don't like this name since it contains fields named `valueSchemaID` and `generatedSchemaID` but at the moment, I can't think of what else to call it

Reply:
I thought of calling it `DerivedSchemaID` but now we have two kinds of generated schemas, the "derived" ones (WC op) and the RMD ones. That's why I called it "generated", although I agree it's not a completely ideal name... IDK what else to call it...

Reply:
My concern is not due to derived vs generated but rather the fact that the name doesn't signify that it is a combination of value schema id and another generated schema id. `ValueAndGeneratedSchemaID` is a possibility, but I don't like this a lot too 😅

Reply:
Generated schema IDs in Venice are composite. They always related to the value schema they were generated from, and then we tack on an extra ID to version the generation logic itself. I could add JavaDoc for that.

Reply:
Yeah, I guess javadoc should help and be enough